tests/numerics/petsc_vector_test.C

```diff
@@ -129,6 +138,89 @@ public:
                               libMesh::TOLERANCE*libMesh::TOLERANCE);
   }
 
+  void testPetscVectorGetSetThread()
+  {
+    LOG_UNIT_TEST;
+
+    Mesh mesh(*TestCommWorld);
+    EquationSystems es(mesh);
+
+    LinearImplicitSystem & sys = es.add_system<LinearImplicitSystem> ("test");
+
+    sys.add_variable("u", FEType(0, MONOMIAL));
+
+    MeshTools::Generation::build_line (mesh,
+                                       50,
+                                       0., 1.,
+                                       EDGE2);
+    es.init();
+
+    *sys.solution = 1.0;
+
+    ConstElemRange active_local_elem_range(mesh.active_local_elements_begin(),
+                                           mesh.active_local_elements_end());
+    std::vector<dof_id_type> cache(mesh.parallel_n_elem());
+    for (const auto & elem : active_local_elem_range)
+    {
+      std::vector<dof_id_type> indices;
+      sys.get_dof_map().dof_indices(elem, indices, 0);
+      libmesh_assert(indices.size() == 1);
+      cache[elem->id()] = indices[0];
+    }
+
+    ConstMonomialGetSetThread test_thread(sys, cache);
+
+    Threads::parallel_for(active_local_elem_range, test_thread);
+
+    sys.solution->close();
+
+    const libMesh::dof_id_type first = sys.solution->first_local_index();
+    const libMesh::dof_id_type last  = sys.solution->last_local_index();
+
+    for (libMesh::dof_id_type n=first; n != last; n++)
+      LIBMESH_ASSERT_FP_EQUAL(libMesh::libmesh_real((*sys.solution)(n)),
+                              libMesh::Real(1.0),
```

Comment:
should this be 0.5?

Reply:
Nice, thanks! I was altering things in there and forgot to reset it back to 0.5.